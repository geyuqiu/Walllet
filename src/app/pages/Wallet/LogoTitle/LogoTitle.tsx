import React from "react";
import {useHistory} from "react-router-dom";
import styled from 'styled-components';

import {SvgCollection} from '../../../assets/svg';

type LogoContainerProps = {
	width: number;
}

const LogoContainer = styled.div<LogoContainerProps>`
  ${({width}) => width && `
    width: ${width}px;
  `};
  height: auto;
`;

const {Logo} = SvgCollection;

export const LogoTitle = () => {
	const history = useHistory();

	return (
		<div className="flex items-center my-auto cursor-pointer" onClick={() => history.push('/fee')}>
			<LogoContainer width={57}>
				<Logo width={44}/>
			</LogoContainer>

			<span className="text-2xl font-bold lg:hidden xl:block" data-testid="logo__text">ARK Wallet</span>
		</div>
	);
};
