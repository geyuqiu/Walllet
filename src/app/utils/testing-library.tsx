import {render} from '@testing-library/react';
import {createMemoryHistory} from "history";
import React from "react";
import {Router} from "react-router-dom";

const customRender = (component: React.ReactElement, options: any = {}) =>
	render(component, {...options});

export const renderWithRouter = (
	component: React.ReactElement,
	{
		routes = ["/"],
		history = createMemoryHistory({ initialEntries: routes })
	} = {},
) => {
	const RouterWrapper = ({ children }: { children: React.ReactNode }) =>
			<Router history={history}>{children}</Router>;

	return {
		...customRender(component, {wrapper: RouterWrapper}),
		history,
	};
};
