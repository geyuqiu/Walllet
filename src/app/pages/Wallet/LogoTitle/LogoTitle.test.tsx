import {fireEvent, render} from "@testing-library/react";
import React from "react";

import {renderWithRouter} from '../../../utils/testing-library';
import {LogoTitle} from "./LogoTitle";

describe("LogoTitle", () => {

	it("should match snapshot", () => {
		const { container } = render(<LogoTitle />);

		expect(container).toMatchSnapshot();
	});

	it("navigates to fee route", () => {
		const {getByTestId, history} = renderWithRouter(<LogoTitle/>);

		fireEvent.click(getByTestId("logo__text"));
		expect(history.location.pathname).toEqual("/fee");
	});
});
