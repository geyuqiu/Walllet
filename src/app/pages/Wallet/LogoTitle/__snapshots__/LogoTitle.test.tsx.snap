// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoTitle should match snapshot 1`] = `
<div>
  <div
    class="flex items-center my-auto cursor-pointer"
  >
    <div
      class="sc-bdfBwQ brMhTG"
      width="57"
    >
      <svg
        width="44"
      >
        logo.svg
      </svg>
    </div>
    <span
      class="text-2xl font-bold lg:hidden xl:block"
      data-testid="logo__text"
    >
      ARK Wallet
    </span>
  </div>
</div>
`;
